KwcNewsletter/Kwc/Newsletter/Detail/Component.php

```diff
@@ -54,7 +54,7 @@ public function removeFromQueue($model = '', $ids = array())
         $newsletter = $this->getData()->row;
         $queueModel = $this->getData()->parent->getComponent()->getChildModel()->getDependentModel('Queues');
         $select = $queueModel->select()
-            ->whereEquals('recipient_model', $model)
+            ->whereEquals('recipient_model_shortcut', $this->getData()->getChildComponent('_mail')->getComponent()->getRecipientModelShortcutFromModel($model))
```

Comment:
hm, diese getRecipientModelShortcutFromModel funktion find ich sub-optimal. Theoretisch kann ein model mehrfach vorkommen.

Könntest du nicht alternativ statt das model zu übergeben (in removeFromQueue und importToQueue) den shortcut übergeben? Ist vermutlich eine größere änderung da du da eventuell auch im javascript das durchschleifen musst? aber ich würds trotzdem besser finden...

Reply:
wenn wir das gleiche model für mehrere recipientsources verwenden haben wir generell ein problem seh ich grad
hab all unsere newsletterwebs durchgesehen den fall gabs bissher noch nie??

Reply:
ich erklärs dir dann nächste woche was ich mein ;) das macht so wirklich keinen sinn XD

Reply:
@nsams ist gefixed